hail/src/main/scala/is/hail/expr/ir/agg/AggregatorState.scala

```diff
@@ -203,41 +196,38 @@ class PrimitiveRVAState(val vtypes: Array[VirtualTypeWithReq], val kb: EmitClass
   }
 
   def store(cb: EmitCodeBuilder, regionStorer: (EmitCodeBuilder, Value[Region]) => Unit, destc: Code[Long]): Unit = {
-    foreachField({
-      case (i, (None, v, t)) =>
-        cb += Region.storePrimitive(t, storageType.fieldOffset(destc, i))(v)
-      case (i, (Some(m), v, t)) =>
-        val dest = cb.newLocal("prim_rvastate_store_dest", destc)
-        cb.ifx(m,
-          cb += storageType.setFieldMissing(dest, i),
-          {
-            cb += storageType.setFieldPresent(dest, i)
-            cb += Region.storePrimitive(t, storageType.fieldOffset(dest, i))(v)
-          })
-    })
+    // everything is primitives, so this is fine! :)
+    val r = new Value[Region] {
+      def get: Code[Region] = Code._null[Region]
+    }
+    val dest = cb.newLocal("prim_rvastate_store_dest", destc)
+    storageType.storeAtAddressFromFields(cb, dest, r, fields.map(_.load), false)
```

Comment:
can you just pass a `null` `Code[Region]` instead of a wrapped null?

Reply:
We'll try it.

Reply:
seems to work